cmd/kubeadm/app/cmd/phases/controlplanepreparejoin.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
+	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	certsphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/certs"
+	controlplanephase "k8s.io/kubernetes/cmd/kubeadm/app/phases/controlplane"
+	etcdphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/etcd"
+	kubeconfigphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/kubeconfig"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+)
+
+type controlPlanePrepareData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+}
+
+// NewControlPlanePreparePhase creates a kubeadm workflow phase that implements the preparation of the node to serve a control plane
+func NewControlPlanePreparePhase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "control-plane-prepare",
+		Short: "Prepares the machine for serving a control plane.",
+		Long:  cmdutil.MacroCommandLongDescription,
+		Phases: []workflow.Phase{
+			newCertsSubphase(),
+			newKubeconfigSubphase(),
+			newManifestsSubphase(),
+		},
+	}
+	return phase
+}
+
+func newCertsSubphase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "certs",
+		Short: "Generates the certificates for all the new control plane components",
+		Run:   runCertsPhaseLocal(),
+	}
+	return phase
+}
+
+func newKubeconfigSubphase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "kubeconfig",
+		Short: "Generates the kubeconfig for all new control plane components",
+		Run:   runKubeconfigPhaseLocal(),
+	}
+	return phase
+}
+
+func newManifestsSubphase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "manifests",
+		Short: "Generates the manifests for all the new control plane components",
+		Run:   runManifestsPhaseLocal(),
+	}
+	return phase
+}
+
+func runCertsPhaseLocal() func(c workflow.RunData) error {
```

Comment:
I guess that we just need to declare the functions instead of returning this.

Reply:
Ugh, yes.